pilot/pkg/model/context.go

```diff
@@ -267,10 +268,15 @@ func (node *Proxy) SetServiceInstances(env *Environment) error {
 }
 
 func (node *Proxy) SetWorkloadLabels(env *Environment) error {
+	// The WorkloadLabels is already parsed from Node metadata["istio.io/metadata"][labels]
+	if node.WorkloadLabels != nil {
```

Comment:
what if labels update?

Reply:
good question, for k8s it can be updated in `Discovery.WorkloadUpdate`. But for other registries, we have no way to update it currently.  And i do not want to make pilot model care much about a specific platform.

Reply:
Yeah so won't this will break kubernetes label updates, because if they are set it will short circuit here if they are already set?

Reply:
Please update the comment. It is no longer under `istio.io/metadata`

Reply:
>Yeah so won't this will break kubernetes label updates, because if they are set it will short circuit here if they are already set?

Yep, see https://github.com/istio/istio/pull/16124/files#diff-6713acfe7b9e389e7dcd429d08bee859R480

Reply:
done